tool-map.nix

```diff
@@ -41,6 +41,13 @@ compiler-nix-name: tool: {
       # `tool` normally ignores the `cabal.project` (if there is one in the hackage source).
       # We need to use the github one (since it has settings to make hls build).
       cabalProject = __readFile (src + "/cabal.project");
+      configureArgs = "--disable-benchmarks --disable-tests";
+      modules = [{
+        packages.ghcide.patches = [
+          # https://github.com/haskell/haskell-language-server/issues/4046#issuecomment-1926242056
```

Comment:
I've only got  this working with hls-2.6 (2.7 definitely  did not work),  but I  suspect it  won't work with hls <= 2.4. Have you tested on 2.2 and 2.4?